lib/spritely/sprockets/transformer.rb

```diff
@@ -0,0 +1,41 @@
+require 'spritely/version'
+require 'spritely/sprite_map'
+
+module Spritely
+  module Sprockets
+    class Transformer < Struct.new(:input)
+      def self.call(input)
+        new(input).call
+      end
+
+      def self.cache_key
+        @cache_key ||= "#{name}:#{Spritely::VERSION}".freeze
+      end
+
+      def call
+        sprite_map = SpriteMap.new(input[:name].remove("sprites/"), input[:environment], input[:metadata][:sprite_directives])
+
+        data = cache.fetch([self.class.cache_key, input[:name], sprite_map.cache_key]) do
+          sprite_map.files.each do |file|
+            context.depend_on(File.dirname(file))
```

Comment:
It might be possible for us to accomplish this in the `Preprocessor`, following the pattern in [`Sprockets::DirectiveProcessor`](https://github.com/rails/sprockets/blob/v3.5.2/lib/sprockets/directive_processor.rb#L291-L301).